src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -179,12 +178,13 @@ impl<S: Store> IndexScanExecutor<S> {
 impl<S: Store> Executor for IndexScanExecutor<S> {
     fn next(&mut self) -> Result<Option<Row>> {
         let now = Instant::now_coarse();
```

Comment:
Guard is not used as this way, you can take a look at https://github.com/tikv/tikv/blob/9b9766895ccf3671d8aed464ff3d7023c3d091d2/src/pd/client.rs#L92 for an example. We create a guard at the beginning, and the time will be recorded at every exit (thanks to guard's Drop).

Reply:
I'm curious about where to store the Duration. If we just store in the returned Guard then we need to copy it to `metrics` in every exit point.

Reply:
I use `pub runtime_stats: &'a mut RuntimeStats,` it should be the same idea with you. But the only way to put the guard is before the `return`. Place the guard at the beginning will lead to second mutable borrows error because `self.get_row_from_range_scanner()` is a mutable borrow. `self.get_row_from_point(range)` is also a mutable borrow. So we can only place the guard in the `if` statement.

Reply:
Fine, this is forbidden by the rust borrow rule. Then looks like this guard is useless because it does not guard anything 🤣 How about extracting the main logic (which has multiple exit places) into another function, and make a single function call here (so there is only 1 exit). Then use the traditional duration here? The core idea behind these moves (including using guards) are making sure that we will not forget the duration when there is more exit points added in future.

Reply:
How about write another function `_next` to call `next`, then add the duration after calling `next`. But the guard should also place after calling `next` since it needs mutable borrow.